OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -351,74 +395,63 @@ Target ChooseTarget(Actor self, AttackBase ab, PlayerRelationship attackStances,
 			var chosenTargetPriority = int.MinValue;
 			var chosenTargetRange = 0;
 
-			var activePriorities = activeTargetPriorities.ToList();
-			if (activePriorities.Count == 0)
+			if (!HasEnabledTargetPriorities())
 				return chosenTarget;
 
-			var targetsInRange = self.World.FindActorsInCircle(self.CenterPosition, scanRange)
-				.Select(Target.FromActor);
+			// PERF: Avoid early conversion to Target.
+			var targetsInRange = new List<Target>(64);
```

Comment:
This still shows up everywhere in allocations. List<__Canon>..ctor(int). Maybe we should move this List to class scope and clear it at the end of the function?

64 is also quite a large number for a more rarer case